### ISSUE TITLE
ValueError using GLPK

### ISSUE THREAD

**Issue by MGlauer** [openego/eTraGo#59]
Running eTraGo with `solver=glpk` and otherwise default settings on the most recent oedb data yields
`ValueError: Expecting 's' row after 'c' rows`
```
Traceback (most recent call last):
  File "[path]/eTraGo/etrago/appl.py", line 170, in <module>
    network = etrago(args)
  File "[path]/eTraGo/etrago/appl.py", line 136, in etrago
    network.lopf(scenario.timeindex, solver_name=args['solver'], extra_functionality=extra_functionality)
  File "[path]/eTraGo/venv/src/pypsa/pypsa/opf.py", line 1099, in network_lopf
    keepfiles=keep_files, options=solver_options)
  File "[path]/eTraGo/venv/lib/python3.5/site-packages/pyomo/opt/base/solvers.py", line 610, in solve
    result = self._postsolve()
  File "[path]/eTraGo/venv/lib/python3.5/site-packages/pyomo/opt/solver/shellcmd.py", line 268, in _postsolve
    results = self.process_output(self._rc)
  File "[path]/eTraGo/venv/lib/python3.5/site-packages/pyomo/opt/solver/shellcmd.py", line 330, in process_output
    self.process_soln_file(results)
  File "[path]/eTraGo/venv/lib/python3.5/site-packages/pyomo/solvers/plugins/solvers/GLPK.py", line 362, in process_soln_file
    raise ValueError(msg)
ValueError: Error parsing solution data file, line 
```

It seems to fail at the very first line of the *.glpk.glp-file, which is:
`p lp min 5641 4090 12292`

Additionally, I receive the following warnings:

```
No data for StoragePqSet in column p_set.
Series p_set of component StorageUnit could not be imported
[path]/eTraGo/etrago/extras/utilities.py:300: RuntimeWarning: divide by zero encountered in reciprocal
  'r_pu': lambda x: np.reciprocal(np.sum(np.reciprocal(x))),
[path]/eTraGo/etrago/extras/utilities.py:314: RuntimeWarning: divide by zero encountered in reciprocal
  'x_pu': lambda x: np.reciprocal(np.sum(np.reciprocal(x))),
[path]/eTraGo/etrago/extras/utilities.py:299: RuntimeWarning: divide by zero encountered in reciprocal
  'r': lambda x: np.reciprocal(np.sum(np.reciprocal(x))),
```

**Comment by mariusves** [comment]
I don't have a solution for the glpk bug, but I can tell you that I'm working on a fix to get rid of the warnings.

**Comment by ulfmueller** [comment]
By the way, considering the warning,  why do we have x values which are 0 in the dataset? - I thought that was not possible anymore. r - ok but x should not be possible anyhow because the lopf would devide by 0 then as well. Are there any branches (transformer/lines) which have x=0?

**Comment by ulfmueller** [comment]
@MGlauer I could not reproduce your error. I just successfully optimized using the glpk solver. I simplified a bit the default values for the args. May be pull the [changes](https://github.com/openego/eTraGo/commit/ff4f6ebf3eeb021cd6a4496d71dce3893c96cc84) and try out again.

**Comment by mariusves** [comment]
We don't have `x` values that are `0` but `x_pu` values that are `0`. That is because the grouping has to be done before the calculations and before PyPSA calculates the pu-values. 

For `r` values, these are not always available for certain lines, hence the warning.

I'm going to implement a check for zeros in the grouping function to prevent these kind of warnings.

**Comment by MGlauer** [comment]
Unfortunately, the error remains.  Even a compete re-installation of eTraGo with a fresh venv did not fix anything. From my perspective there are two potential points of failure:

1. You are using a different data set than I am. I used the most recent data from the oedb. 
2. You are using a different version for some package which is not listed in the `setup.py`. You could just mail me a `pip freeze` and I could check myself.
3. There is still something of in my configuration. A snapshot of your configuration using glpk might be helpful as well. 

I am working with GLPK LP/MIP Solver, v4.57

**Comment by ulfmueller** [comment]
> 1. You are using a different data set than I am. I used the most recent data from the oedb. 

Nope. Same here.
> 2. You are using a different version for some package which is not listed in the `setup.py`. You could 
just mail me a `pip freeze` and I could check myself.

done. 
> 3. There is still something of in my configuration. A snapshot of your configuration using glpk might be helpful as well. 

do you mean the args[]? Like I posted, they were [these](https://github.com/openego/eTraGo/commit/ff4f6ebf3eeb021cd6a4496d71dce3893c96cc84)

> I am working with GLPK LP/MIP Solver, v4.57

same here.

**Comment by BartelsJ** [comment]
I reproduced the error and got rid of it turning load_shedding to True (see closed issue #16). Interestingly, the error disappeared when I switched the load_shedding back to False. It seems that the bug is caused by a failing definition of marginal_costs. When I defined them explicit like "marginal_cost = 1000" the calculation runs.

**Comment by MGlauer** [comment]
Setting `load_shedding` did not help either. **But**: I finally found the origin of the error. This is actually a [bug in pyomo 5.2](https://github.com/Pyomo/pyomo/issues/146). Installing the master branch-version solved the issue.
